test/backend/db_tests.js

```diff
@@ -450,6 +451,49 @@ module.exports = function(config, DB) {
         )
 
         test(
+          'locked accounts',
+          function (t) {
+            t.plan(5)
+
+            var lockedAt = Date.now()
+            var unlockCode = hex16()
+            var uid = Buffer(ACCOUNT.uid)
+            var email = Buffer(ACCOUNT.email)
+
+              // set lockedAt
+            return db.lockAccount(uid, { lockedAt: lockedAt, unlockCode: unlockCode })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .lockAccount() request' + String(err))
+              })
+              .then(function(result) {
+                t.deepEqual(result, {}, 'Returned an empty object for lockAccount')
+                return db.account(uid)
+              })
+              .then(function(account) {
+                t.equal(account.lockedAt, lockedAt, 'account should now be locked')
+                return db.unlockCode(uid)
+              })
+              .then(function (_unlockCode) {
+                t.deepEqual(_unlockCode.unlockCode, unlockCode, 'unlockCode should be set')
+
+                // try to unlock the account
+                return db.unlockAccount(uid)
+              })
+              .then(null, function(err) {
+                t.fail('We should not have failed this .unlockAccount() request' + String(err));
```

Comment:
I think you're right about being more explicit and returning the error to the client for the purposes of error reporting to the user. Let's leave this as you have it already.